### ISSUE TITLE
Crash due to server usage of client library

### ISSUE THREAD

**Issue by ViRb3** [Qelifern/IronFurnaces#143]
### Iron Furnaces Version

3.6.4, Forge

### Describe your issue

Enabling the Comparator on any Iron Furnace crashes the server.

It seems like in version 3.6.4, which as far as I can see is not on GitHub, you changed the `BlockIronFurnaceBase.getComparatorInputOverride` method to use Apache Commons, which is a library present only in the client, but not server. So, when this method is called on the server, it crashes. Reverting to 3.6.3 appears to fix the problem. Decompiled code that crashes:

<img width="664" alt="Screenshot 2022-12-30 at 23 41 18" src="https://user-images.githubusercontent.com/2650170/210113480-5c786fc8-3b1a-49bc-96ca-63d92a24ea8c.png">

To fix, can just use `new ArrayList<>` instead of `Lists.newArrayList()` I reckon.

### Crash Report

```
java.lang.NoClassDefFoundError: org/apache/commons/compress/utils/Lists
	at ironfurnaces.blocks.furnaces.BlockIronFurnaceBase.getComparatorInputOverride(BlockIronFurnaceBase.java:296) ~[ironfurnaces-1.19.2-3.6.4.jar%23245!/:3.6.4] {re:classloading,pl:runtimedistcleaner:A}
	at ironfurnaces.blocks.furnaces.BlockIronFurnaceBase.calculateOutput(BlockIronFurnaceBase.java:370) ~[ironfurnaces-1.19.2-3.6.4.jar%23245!/:3.6.4] {re:classloading,pl:runtimedistcleaner:A}
	at ironfurnaces.blocks.furnaces.BlockIronFurnaceBase.m_6376_(BlockIronFurnaceBase.java:410) ~[ironfurnaces-1.19.2-3.6.4.jar%23245!/:3.6.4] {re:classloading,pl:runtimedistcleaner:A}
	at net.minecraft.world.level.block.state.BlockBehaviour$BlockStateBase.m_60775_(BlockBehaviour.java:578) ~[server-1.19.2-20220805.130853-srg.jar%23404!/:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:APP:kubejs-common.mixins.json:BlockStateBaseMixin,pl:mixin:APP:radon.mixins.json:block.MixinAbstractBlockState,pl:mixin:APP:immersiveengineering.mixins.json:coremods.AbstractBlockStateMixin,pl:mixin:APP:byg.mixins.json:common.block.MixinBlockBehaviourBlockStateBase,pl:mixin:APP:ferritecore.blockstatecache.mixin.json:BlockStateBaseMixin,pl:mixin:A}
	at ironfurnaces.tileentity.furnaces.BlockIronFurnaceTileBase.tick(BlockIronFurnaceTileBase.java:571) ~[ironfurnaces-1.19.2-3.6.4.jar%23245!/:3.6.4] {re:classloading}
	at net.minecraft.world.level.chunk.LevelChunk$BoundTickingBlockEntity.m_142224_(LevelChunk.java:687) ~[server-1.19.2-20220805.130853-srg.jar%23404!/:?] {re:classloading,pl:accesstransformer:B}
	at net.minecraft.world.level.chunk.LevelChunk$RebindableTickingBlockEntityWrapper.m_142224_(LevelChunk.java:780) ~[server-1.19.2-20220805.130853-srg.jar%23404!/:?] {re:classloading,pl:accesstransformer:B}
	at net.minecraft.world.level.Level.redirect$bdp000$redirectTick(Level.java:3038) ~[server-1.19.2-20220805.130853-srg.jar%23404!/:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:APP:kubejs-common.mixins.json:LevelMixin,pl:mixin:APP:lootr.mixins.json:MixinLevel,pl:mixin:APP:utilitix.mixins.json:MixinLevel,pl:mixin:APP:itshallnottick.mixins.json:EntityTickMixin,pl:mixin:APP:observable.common.json:LevelMixin,pl:mixin:A}
	at net.minecraft.world.level.Level.m_46463_(Level.java:446) ~[server-1.19.2-20220805.130853-srg.jar%23404!/:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:APP:kubejs-common.mixins.json:LevelMixin,pl:mixin:APP:lootr.mixins.json:MixinLevel,pl:mixin:APP:utilitix.mixins.json:MixinLevel,pl:mixin:APP:itshallnottick.mixins.json:EntityTickMixin,pl:mixin:APP:observable.common.json:LevelMixin,pl:mixin:A}
	at net.minecraft.server.level.ServerLevel.m_8793_(ServerLevel.java:331) ~[server-1.19.2-20220805.130853-srg.jar%23404!/:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:APP:kubejs-common.mixins.json:ServerLevelMixin,pl:mixin:APP:supplementaries-common.mixins.json:ServerLevelMixin,pl:mixin:APP:pluto.mixins.json:server.fastchunkentityaccess.ServerWorldMixin,pl:mixin:APP:immersiveengineering.mixins.json:coremods.ServerWorldMixin,pl:mixin:APP:byg.mixins.json:common.world.MixinServerLevel,pl:mixin:APP:ad_astra-common.mixins.json:ServerLevelMixin,pl:mixin:APP:observable.common.json:ServerLevelMixin,pl:mixin:APP:create.mixins.json:accessor.ServerLevelAccessor,pl:mixin:A}
	at net.minecraft.server.MinecraftServer.m_5703_(MinecraftServer.java:866) ~[server-1.19.2-20220805.130853-srg.jar%23404!/:?] {re:mixin,pl:accesstransformer:B,re:computing_frames,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:APP:kubejs-common.mixins.json:MinecraftServerMixin,pl:mixin:APP:kubejs-common.mixins.json:inject_resources.MinecraftServerMixin,pl:mixin:APP:balm.mixins.json:MinecraftServerMixin,pl:mixin:APP:structure_gel.mixins.json:MinecraftServerMixin,pl:mixin:APP:ae2.mixins.json:spatial.MinecraftServerMixin,pl:mixin:APP:byg.mixins.json:access.MinecraftServerAccess,pl:mixin:APP:byg.mixins.json:server.MixinMinecraftServer,pl:mixin:APP:elementalcraft.mixins.json:MixinMinecraftServer,pl:mixin:A}
	at net.minecraft.server.dedicated.DedicatedServer.m_5703_(DedicatedServer.java:292) ~[server-1.19.2-20220805.130853-srg.jar%23404!/:?] {re:mixin,pl:accesstransformer:B,re:computing_frames,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:APP:nochatreports.mixins.json:server.MixinDedicatedServer,pl:mixin:APP:tombstone.mixins.json:DedicatedServerMixin,pl:mixin:A}
	at net.minecraft.server.MinecraftServer.m_5705_(MinecraftServer.java:806) ~[server-1.19.2-20220805.130853-srg.jar%23404!/:?] {re:mixin,pl:accesstransformer:B,re:computing_frames,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:APP:kubejs-common.mixins.json:MinecraftServerMixin,pl:mixin:APP:kubejs-common.mixins.json:inject_resources.MinecraftServerMixin,pl:mixin:APP:balm.mixins.json:MinecraftServerMixin,pl:mixin:APP:structure_gel.mixins.json:MinecraftServerMixin,pl:mixin:APP:ae2.mixins.json:spatial.MinecraftServerMixin,pl:mixin:APP:byg.mixins.json:access.MinecraftServerAccess,pl:mixin:APP:byg.mixins.json:server.MixinMinecraftServer,pl:mixin:APP:elementalcraft.mixins.json:MixinMinecraftServer,pl:mixin:A}
	at net.minecraft.server.MinecraftServer.m_130011_(MinecraftServer.java:654) ~[server-1.19.2-20220805.130853-srg.jar%23404!/:?] {re:mixin,pl:accesstransformer:B,re:computing_frames,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:APP:kubejs-common.mixins.json:MinecraftServerMixin,pl:mixin:APP:kubejs-common.mixins.json:inject_resources.MinecraftServerMixin,pl:mixin:APP:balm.mixins.json:MinecraftServerMixin,pl:mixin:APP:structure_gel.mixins.json:MinecraftServerMixin,pl:mixin:APP:ae2.mixins.json:spatial.MinecraftServerMixin,pl:mixin:APP:byg.mixins.json:access.MinecraftServerAccess,pl:mixin:APP:byg.mixins.json:server.MixinMinecraftServer,pl:mixin:APP:elementalcraft.mixins.json:MixinMinecraftServer,pl:mixin:A}
	at net.minecraft.server.MinecraftServer.m_206580_(MinecraftServer.java:244) ~[server-1.19.2-20220805.130853-srg.jar%23404!/:?] {re:mixin,pl:accesstransformer:B,re:computing_frames,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:APP:kubejs-common.mixins.json:MinecraftServerMixin,pl:mixin:APP:kubejs-common.mixins.json:inject_resources.MinecraftServerMixin,pl:mixin:APP:balm.mixins.json:MinecraftServerMixin,pl:mixin:APP:structure_gel.mixins.json:MinecraftServerMixin,pl:mixin:APP:ae2.mixins.json:spatial.MinecraftServerMixin,pl:mixin:APP:byg.mixins.json:access.MinecraftServerAccess,pl:mixin:APP:byg.mixins.json:server.MixinMinecraftServer,pl:mixin:APP:elementalcraft.mixins.json:MixinMinecraftServer,pl:mixin:A}
	at java.lang.Thread.run(Thread.java:831) [?:?] {re:mixin}
Caused by: java.lang.ClassNotFoundException: org.apache.commons.compress.utils.Lists
	at jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:636) ~[?:?] {}
	at java.lang.ClassLoader.loadClass(ClassLoader.java:519) ~[?:?] {}
	at cpw.mods.cl.ModuleClassLoader.loadClass(ModuleClassLoader.java:137) ~[securejarhandler-2.1.4.jar:?] {}
	at java.lang.ClassLoader.loadClass(ClassLoader.java:519) ~[?:?] {}
	at cpw.mods.cl.ModuleClassLoader.loadClass(ModuleClassLoader.java:137) ~[securejarhandler-2.1.4.jar:?] {}
	at java.lang.ClassLoader.loadClass(ClassLoader.java:519) ~[?:?] {}
	... 16 more
```

**Comment by Qelifern** [comment]
Fixed in 1.19.3-3.7.2

**Comment by Swedz** [comment]
Please fix this in a 1.19.2 version as well.

For those who are looking for a quick workaround, use 1.19.2-3.6.3 on the curseforge page. When installed on a server, it seems to be compatible with clients running 3.6.4, and nothing seems to be too seriously problematic from what I can tell.

**Comment by mcmso** [comment]
> Please fix this in a 1.19.2 version as well.

Please fix this in a 1.18.2 version as well.

**Comment by mcmso** [comment]
> > Please fix this in a 1.19.2 version as well.
> 
> Please fix this in a 1.18.2 version as well.
[crash-2023-01-07_23.08.27-server.txt](https://github.com/Qelifern/IronFurnaces/files/10366575/crash-2023-01-07_23.08.27-server.txt)

**Comment by Qelifern** [comment]
Fixed in 1.19.2 3.6.5

**Comment by LindaJuffermans** [comment]
Can you provide the fix for 1.18.2 as well, please?

[crash-2023-01-29_08.01.33-server.txt](https://github.com/Qelifern/IronFurnaces/files/10532862/crash-2023-01-29_08.01.33-server.txt)

**Comment by DemonMartin** [comment]
Please provide a fix for 1.18.2 too, as I cannot fix my server besides uploading a backup...

**Comment by LindaJuffermans** [comment]
> Please provide a fix for 1.18.2 too, as I cannot fix my server besides uploading a backup...

You can workaround it by replacing the furnace either in the NBT data or by using a repeating command block that does
`setblock x y z pink_wool`
Then slowly approach the area.

**Comment by DemonMartin** [comment]
> > Please provide a fix for 1.18.2 too, as I cannot fix my server besides uploading a backup...
> 
> You can workaround it by replacing the furnace either in the NBT data or by using a repeating command block that does `setblock x y z pink_wool` Then slowly approach the area.

I tried replacing the block tho it doesn‘t seem to fix it cuz the server crashes nearly instantly after it fully started

**Comment by ViRb3** [comment]
@DemonMartin Use Amulet to replace/delete the block — https://www.amuletmc.com/

**Comment by DemonMartin** [comment]
@ViRb3 
Sadly, this doesn't seem to work as it doesn't show me the right dimension...
The dimension is "Level dimension: ftbsbc:team/cockles-sundaes-semilog" based on the error, but I can't seem to find it there

**Comment by EvanHsieh0415** [comment]
@Qelifern Will there still be a fix for this bug on 1.18.2?
This problem also occurs in ATM7-To The Sky.

**Comment by shinji257** [comment]
Can we get this reopened and the fix backported to 1.18.2?  Currently affecting Project: Phoenix Reborn and 3.3.3 is the last version released.  I'd try to do it myself but the code updates for 3.6.4 and 3.6.5 on 1.19.2 have not been published to Github in order to allow us to see what the fix was.

**Comment by LionZXY** [comment]
Patched version of the mod for 1.18.2. 

[ironfurnaces-1.18.2-3.3.3-patched.jar.zip](https://github.com/Qelifern/IronFurnaces/files/13954266/ironfurnaces-1.18.2-3.3.3-patched.jar.zip)

Just unzip and copy this jar to the mods folder and delete the original file `ironfurnaces-1.18.2-3.3.3.jar`. It should **ONLY** be done on the server
Warning: installing mods from third party sources is dangerous, don't trust anyone, including me

You can build such a .jar on your own:
1) Download the mod file from [curseforge](https://www.curseforge.com/minecraft/mc-mods/iron-furnaces/download/4183356)
2) Download the Apache Commons Compress library file from [maven](https://repo1.maven.org/maven2/org/apache/commons/commons-compress/1.25.0/commons-compress-1.25.0.jar)
3) Extract the .jar mod file as a .zip archive
4) Create a folder in the unzipped mod file at the path org/apache/commons/compress/utils
5) Unzip the .jar library file as a .zip archive and copy org/apache/commons/compress/utils/Lists.class to org/apache/commons/compress/utils/ of the unzipped mod folder.


The result should look like this:
![image](https://github.com/Qelifern/IronFurnaces/assets/5871715/582fe246-e099-4d31-84dd-1e3eab271573)

6) Compress the mod file back to .zip and rename the extension at the end to .jar

cc: https://github.com/FTBTeam/FTB-Modpack-Issues/issues/3166
